### ISSUE TITLE
[Beta 6.21] Feedback from my blind playthrough, Part 3

### ISSUE THREAD

**Issue by lOmicronl** [Azanor/thaumcraft-beta#1291]
Hello Azanor, I'm currently in my first, "blind" run of Thaumcraft 6. I had previously played 2, 3 and 4, but not 5. And even though it's still in beta, I can already say that version 6 might be my favorite Thaumcraft ever. 2 was really hard to beat in my book, but 6 makes all the right callbacks while combining it with all the great new ideas you've had since then. It's brilliant.

While playing, I've made a list of various things that struck me as odd or wrong. I realize the mod is not done, and that some (or even all) of the things I'm going to remark on are simply down to that. I'm still going to write down all of it, in hopes that at least something will help you in some form or other. Gonna split it up, though, to keep the issue thread from being too unfocused.

This is part 3 (the big one): Flux and Taint.

The Thaumonomicon casually references taint and vitium essentia all over the place, as if it expects me to know exactly what it is right from the start. Problem: I've never seen as much as a single tendril. I was wondering if it might come out of the rift, but it didn't - or if it did, I simply missed it (which is something that should be avoided, game design wise, when it comes to progress-gating elements). I've explored quite a bit, too. Three vanilla maps worth at zoom level 3, so 64x64x3 chunks. No natural taint anywhere I can see. And the Thaumonomicon gives zero guidance - I suppose the Taint entry will show up once I actually find it, which is logical on one hand but doesn't steer the player toward that discovery at all.

Right now, I am gated by my lack of Taint availability on the following research subjects:

- Flux Cleanup: While it perhaps makes thematic sense, it feels a bit like the mod is pulling a rug out under me. There's this big anticipation buildup with experimenting with the flux levels, there's the sense of panic when that rift shows up, there's the sudden dread of realization that your own base is pretty saturated with flux already... it's really good! And then you proceed to the cleanup research and... well. it asks something you've never seen, while giving you every indication that you should already have it. Both in the flavor text, and in the way it asks you to cough up vitium essentia without any further ado. As a player who was trying to rush towards finding a solution, this stopped me dead and felt really bad. Perhaps instead of the requirement of having vitium essentia on hand, the requirement could say "study flux" or even "examine something vile", in the same way that for example the Mirror Magic research asks you to "study teleportation". Another step could then ask for distilling that vile thing into vitium essentia to continue the research as normal.

- Focus Effect: Flux: Again, the mod makes the assumption that the player knows that taint/vitium equals flux, and this research node is always visible and unlocked simply by completing the elemental effects without any extra checks. It would be far more elegant if the whole research node was hidden, and unlocked only when the flux entry in the Thaumonomicon has been completed. After all, up until that point, the flavor is that the player doesn't know what flux is, so why would he go and formulate a focus effect derived from it? Of course, I realize that there are very juicy focus effects gated behind it, but I don't see a real problem with them too being hidden until the flux effect is revealed.

- Essentia Tubes: Kind of a pain, because literally half the Alchemy tree is gated behind them. Okay, it makes sense to proof the tubes against the most exotic forms of essentia. But it again falls into the trap of being available for research before the player even knows what taint and flux are. I'd recommend simply removing the vitium essentia requirement - perhaps replacing it with something similarly rare, like Permutatio (god knows I have no idea where to even get that in Thaumcraft 6!). And then, based on my blind guess that the point of the Thaumium Essentia Smelter centers on not producing as much flux as the standard one, that research would have a requirement to know about flux. It, too, could be hidden until the player has gained flux knowledge, and it too could have a requirement to study flux - and then, as another step, to produce some vitium essentia using the old, worse smelter so that you can use it to develop the new one.

- Infusion Stabilization: Does this really need to involve vitium essentia? I mean sure, flux is the embodiment of chaos, but the flavor text makes no mention of research in that direction. Instead, it talks about experimenting with redstone on the infusion process. Wouldn't machina essentia and a couple of redstone devices (torch, repeater, comparator) be more in line with the flavor? You're trying to build a "mundane technology solution", after all, so let's involve some mundane technology! Ultimately it's not that big of a problem if vitium is required, but the research should probably be gated by the discovery of flux somehow. Right now it unlocks after an infusion goes belly up, and that's definitely themely, but perhaps another check could be made for flux knowledge?

- I have also unlocked the Bottled Taint research, which talks about being able to replicate taint. Replicate how? I've never seen it! That one should definitely be hidden until it makes sense to become visible. True, I could use it to make myself artificial taint and solve my no-vitium-available problem right now, but if that is the intended use, then the flavor is all wrong. It would need a completely different name and description to indicate that you're trying to synthesize something new and unfamiliar in your quest to better understand flux.

- This whole list does not include instances where the flavor text of various Thaumonomicon entries makes small references to flux and taint. One example would be the main Infusion entry. While it's a good thing to inform the player of the consequences, just saying outright "it might produce flux" is immersion-breaking when flux has yet to even be discovered. In this example, just removing the reference works, as the entry already alludes to "more catastrophic consequences", which works perfectly fine. In other entries, rewording references into more vague (or innocent) hints can work.

**Comment by lOmicronl** [comment]
Playing some more, I see now that there is a non-taint vitium source. Since I am on Hard difficulty, with Ender Zoo making mobs harder and adding some mean ones of its own, I put off going for a nether fortress at first.

I still think that everything I've written about flux knowledge being spread through the book before the discovery is made still applies, though. And if anything, my experience only underlines that you really can't assume players having checked off certain progress steps before getting deeper into Thaumcraft... It would be different if Thaumcraft itself was gated behind various vanilla advancements, but right now it starts with the player's first mining trip. I actually had a casting glove and focus before I even found a single diamond... or even bothered to seriously go look for them, in fact.
